core/trino-main/src/test/java/io/trino/RowPagesBuilder.java

```diff
@@ -147,6 +147,11 @@ public RowPagesBuilder pageBreak()
         return types;
     }
 
+    public Optional<List<Integer>> getHashChannels()
+    {
+        return hashChannels;
```

Comment:
the constructor should be doing defensive copy for `hashChannels`

now the hashChannels value is mutable, and not safe to expose.